### PR TITLE
rework : split AuthenticationRouter into atomic component

### PR DESCRIPTION
**Pull request by capyq**
Extract subcomponents into their own files.

### CODE REVIEW

File: src/components/AuthenticationRouter/AuthenticationRouter.tsx [hunk 1]
```diff
@@ -77,10 +45,10 @@ function AuthenticationRouter({
             alignItems="center"
             direction="column"
         >
-            {userManager.error !== null && (
+            {userManager.error && (
```
Comment (Tristan-WorkGH) [inline src/components/AuthenticationRouter/AuthenticationRouter.tsx:48]: We can have an empty string if we haven't retrieve the error message.

File: src/components/AuthenticationRouter/authenticationType.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    logoutError?: { error: Error };
+    unauthorizedUserInfo?: string;
+};
+
+export type UserManagerState = {
+    instance: UserManager | null;
+    error: string | null;
+};
+
+export interface AuthenticationRouterProps {
```
Comment (Tristan-WorkGH) [inline src/components/AuthenticationRouter/authenticationType.ts:25]: Don't we keep the props near the component?

**Review by Tristan-WorkGH — COMMENTED** [review]
We don't touch `src/utils/AuthService`?

**Comment by capyq** [comment]
> We don't touch `src/utils/AuthService`?

It's a minimal rework of the component; the service will be reworked later.

**Comment by capyq** [comment]
@TheMaskedTurtle, rebase done.

**Review by TheMaskedTurtle — APPROVED** [review]

**Pull request merged**